src/content/docs/dev/general-message-passing/sui/sui-programs.mdx

```diff
@@ -0,0 +1,282 @@
+
+At the core of there are two main package involved in sending a GMP message, these are the [Gateway Package](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/axelar_gateway) and the [Gas Service](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/gas_service).
+
+For all complete Sui GMP implementation check out this [example](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/example)
+
+To interact with the example and send a GMP message checkout this [script](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/sui/gmp.js)
+
+## Overview
+
+Sending a message from Sui involves the following steps:
+
+- Register your transaction with the [relayer discovery service](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/relayer_discovery) via the `register_transaction()` function on the relayer discovery service.
+- Prepare the message via the `prepare_message()` function on the [gateway](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/axelar_gateway).
+- Pay gas via the `pay_gas()` function on the [gas service](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/gas_service).
+- Send the message via the send_message function on the [gateway](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/axelar_gateway).
+
+
+## Sui Gateway
+
+The [Gateway](/learn/network/flow/#gateways) is the core contract that facilitates the sending and receiving of cross-chain messages to other chains via the Axelar Network.
+
+### Gateway Object
+
+A shared object that anyone can access and user to refer to the Gateway package. 
+
+```move
+public struct Gateway has key {
+    id: UID,
+    inner: Versioned,
+}
+```
+
+
+The Gateway facilitates sending and receiving of cross-chain messages to other chains via the Axelar Network.
+
+For sending a GMP message, the `send_message()` function needs to be triggered.
+
+### Send Message
+
+The send_message function triggers your cross-chain message from Sui to another blockchain via the Axelar Network. The send_message function requires a MessageTicket struct to be passed in. To create the MessageTicket you can trigger the [prepare_message](#prepare-message) function.
+
+
+<Callout emoji="💡">
+  The reason this process is in two steps is because the Gateway is an upgrade compatible contract. To ensure minimal roadblocks when upgrading the contract the functionality of the Gateway was broken up so that if the Gateway does get upgraded at some point, applications can always continue to call the logic of the V0 `prepare_message()` function and pass the ticket into the V2 version of the `send_message()`, this will minimize breaking changes when upgrading the contract. 
+</Callout>
+
+```move
+public fun send_message(self: &Gateway, message: MessageTicket) {
+    let value = self.value!(b"send_message");
+    value.send_message(message, VERSION);
+}
+```
+
+
+#### Prepare Message
```

Comment:
maybe mention that the intended behaviour is for apps that wish to send calls to return the message ticket and have their frontend send it for easier upgradability.